app/src/marketplace/containers/ProductPage/CommunityStats.jsx

```diff
@@ -0,0 +1,251 @@
+    const { adminFee, joinPartStreamId } = community || {}
+
+    useEffect(() => {
+        if (totalEarnings) {
+            setStats((prev) => ({
+                ...prev,
+                revenue: {
+                    ...prev.revenue,
+                    value: fromAtto(totalEarnings).toNumber() || 0,
+                    loading: false,
+                },
+            }))
+        }
+    }, [totalEarnings])
+
+    useEffect(() => {
+        setStats((prev) => ({
+            ...prev,
+            subscribers: {
+                ...prev.subscribers,
+                value: subscriberCount || 0,
+                loading: false,
+            },
+        }))
+    }, [subscriberCount])
+
+    useEffect(() => {
+        if (adminFee) {
+            setStats((prev) => ({
+                ...prev,
+                adminFee: {
+                    ...prev.adminFee,
+                    value: (adminFee * 100).toFixed(0),
+                    loading: false,
+                },
+            }))
+        }
+    }, [adminFee])
+
+    useEffect(() => {
+        if (created) {
+            setStats((prev) => ({
+                ...prev,
+                created: {
+                    ...prev.created,
+                    value: new Date(created).toLocaleDateString(),
+                    loading: false,
+                },
+            }))
+        }
+    }, [created])
+
+    useEffect(() => {
+        if (totalEarnings && created && memberCount) {
+            const productAgeMs = Date.now() - new Date(created || 0).getTime()
+            const revenuePerMonth = totalEarnings !== 0 ? (totalEarnings / (productAgeMs / MILLISECONDS_IN_MONTH)) : 0
+            const revenuePerMonthPerMember = memberCount.total > 0 ? (revenuePerMonth / memberCount.total) : 0
+
+            setStats((prev) => ({
+                ...prev,
+                averageRevenue: {
+                    ...prev.averageRevenue,
+                    value: revenuePerMonthPerMember,
+                    loading: false,
+                },
+            }))
+        }
+    }, [totalEarnings, created, memberCount])
+
+    useEffect(() => {
+        const { active } = memberCount || {}
+
+        setStats((prev) => ({
+            ...prev,
+            members: {
+                ...prev.members,
+                value: active || 0,
+                loading: false,
+            },
+        }))
+    }, [memberCount])
+
+    const timeOutId = useRef(null)
+    const resetTimeout = useCallback(() => {
+        clearTimeout(timeOutId.current)
+    }, [])
+
+    const getStats = useCallback(async () => {
+        if (!beneficiaryAddress) { return }
+        try {
+            const result = await getCommunityStats(beneficiaryAddress)
+            if (!isMounted()) {
+                return
+            }
+            setTotalEarnings(result.totalEarnings)
+            setMemberCount(result.memberCount)
+        } catch (e) {
+            // Try again
```

Comment:
what error are you expecting here? should probably check the type of error otherwise you could be suppressing something unexpected like a `ReferenceError`

Reply:
Good point, changed so that we try again only if the response is 404 from API, https://github.com/streamr-dev/streamr-platform/pull/755/commits/3c987b57e16f21a9c0613541f27b42850c4376ce